### PR TITLE
Typing wrapped enzyme/testing-library functions again.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

Due to creating untyped wrappers for enzyme/testing-library to include
context providers automatically, usage of these exported functions were
untyped too, preventing type error detection and completions in IDEs.

This PR is adding back typing information for these and fixes the
uncovered type errors. As testing-library's `get*/query*` functions now
return plain `HTMLElement`s, a `asElement` function was introduced,
merely for casting an element to a specified type. At the same time, a
type check (through `instanceof`) is performed, which functions as a
type assert. If it fails, an error is thrown that aborts the test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/dashboard/BigDisplayModeConfiguration.test.jsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ import Query from 'views/logic/queries/Query';
 import ViewState from 'views/logic/views/ViewState';
 import type { ViewStateMap } from 'views/logic/views/View';
 import BigDisplayModeConfiguration from './BigDisplayModeConfiguration';
+import { asElement } from '../../../../test/wrappedTestingLibrary';
```
Comment (kyleknighted) [inline graylog2-web-interface/src/views/components/dashboard/BigDisplayModeConfiguration.test.jsx:14]: This import could be shortened to

```suggestion
import { asElement } from 'wrappedTestingLibrary';
```

File: graylog2-web-interface/src/components/search/SurroundingSearchButton.test.jsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { cleanup, fireEvent, render } from 'wrappedTestingLibrary';
 import DrilldownContext from 'views/components/contexts/DrilldownContext';
 import SurroundingSearchButton from './SurroundingSearchButton';
 import type { SearchesConfig } from './SearchConfig';
+import { asElement } from '../../../test/wrappedTestingLibrary';
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/search/SurroundingSearchButton.test.jsx:8]: This import could be shortened to
```suggestion
import { asElement } from 'wrappedTestingLibrary';
```

File: graylog2-web-interface/src/views/components/widgets/ReplaySearchButton.test.jsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { createElasticsearchQueryString } from 'views/logic/queries/Query';
 import type { ElasticsearchQueryString, TimeRange } from 'views/logic/queries/Query';
 import DrilldownContext from '../contexts/DrilldownContext';
 import ReplaySearchButton from './ReplaySearchButton';
+import { asElement } from '../../../../test/wrappedTestingLibrary';
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog2-web-interface/src/views/components/widgets/ReplaySearchButton.test.jsx:9]: Damn IntelliJ auto-import. Thanks! :)

File: graylog2-web-interface/test/wrappedTestingLibrary.jsx [hunk 4]
```diff
@@ -1,14 +1,26 @@
-import { render } from '@testing-library/react';
+// @flow strict
+import { type Element } from 'react';
+import { render, type RenderOptionsWithoutCustomQueries } from '@testing-library/react';
```
Comment (kyleknighted) [inline graylog2-web-interface/test/wrappedTestingLibrary.jsx:3]: May need to update the import to use the themed version

```suggestion
import { render, type RenderOptionsWithoutCustomQueries } from 'wrappedTestingLibrary';
```

**Pull request merged**